From 576f6e063012935d6336077c27a2e6b63c84e377 Mon Sep 17 00:00:00 2001
From: cosolom <cosolom@1816882.no-reply.drupal.org>
Date: Mon, 20 Mar 2017 17:02:12 +0200
Subject: [PATCH] Issue #2760317 by Ramdas Gaikar, cosolom, m.abdulqader:
 MissingKeyMapError

---
 locationmap.module | 54 +++++++++++++++++++++++++++++++++++++++++++++---------
 1 file changed, 45 insertions(+), 9 deletions(-)

diff --git a/locationmap.module b/locationmap.module
index 4976dc2..1ac58cc 100755
--- a/locationmap.module
+++ b/locationmap.module
@@ -6,10 +6,10 @@
 function locationmap_help($path, $arg) {
   switch ($path) {
     case 'admin/help#locationmap':
-      $output = '<p>' . t('The Location Map module displays a map of your geographic location with Google Maps. The @map_page for the map is automatically generated. A block is also created that displays a thumbnail of the map, linking to the map page. This can be assigned to a page region on the @block_admin_page.',
-        array(
-          '@map_page' => l(t('map page'), 'locationmap'),
-          'block_admin_page' => l(t('block administration page'), 'admin/structure/block')
+     $output = '<p>' . t('The Location Map module displays a map of your geographic location with Google Maps. The !map_page for the map is automatically generated. A block is also created that displays a thumbnail of the map, linking to the map page. This can be assigned to a page region on the !block_admin_page',
+         array(
+          '!map_page' => l(t('map page'), 'locationmap'),
+          '!block_admin_page' => l(t('block administration page'), 'admin/structure/block')
         )) . '.</p>';
       return $output;
   }
@@ -62,6 +62,15 @@ function locationmap_admin_settings($form, &$form_state) {
 
   drupal_add_css($path . '/locationmap_admin.css', $locationmap_admin_css_options);
 
+  $form['locationmap_api_key'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Google Map API Key'),
+    '#description' => t('Enter the Google Map API Key. You can get Google Map API key from !google_api_console',array('!google_api_console' => l(t('Google API Console'), 'https://developers.google.com/maps/documentation/javascript/get-api-key'),
+      )),
+    '#default_value' =>  variable_get('locationmap_api_key', ''),
+    '#required' => TRUE,
+  );
+
   $defaults = array(
     'value' => '',
     'format' => filter_default_format(),
@@ -83,8 +92,8 @@ function locationmap_admin_settings($form, &$form_state) {
     '#type' => 'textfield',
     '#title' => t('Title'),
     '#default_value' => variable_get('locationmap_title', t('Our Location')),
-    '#description' => t("The title of the automatically generated @map_page.",
-      array('@map_page' => l(t('map page'), 'locationmap'))),
+    '#description' => t("The title of the automatically generated !map_page.",
+      array('!map_page' => l(t('map page'), 'locationmap'))),
   );
 
   $form['info']['locationmap_address'] = array('#type' => 'textfield',
@@ -146,7 +155,7 @@ function locationmap_admin_settings($form, &$form_state) {
     '#type' => 'fieldset',
     '#title' => t('Geographical coordinates'),
     '#collapsible' => FALSE,
-    '#description' => t('Geographical coordinates for your location. Location map will try to obtain this information from Google using the address above. You are also able to fine-tune this by dragging the marker on the @map_page. Under normal circumstances you would not set these coordinates manually.', array('@map_page' => l(t('map page'), 'locationmap'))),
+    '#description' => t('Geographical coordinates for your location. Location map will try to obtain this information from Google using the address above. You are also able to fine-tune this by dragging the marker on the !map_page. Under normal circumstances you would not set these coordinates manually.', array('!map_page' => l(t('map page'), 'locationmap'))),
   );
 
   $form['info']['latlng']['locationmap_lat'] = array(
@@ -265,7 +274,19 @@ function locationmap_page() {
 
   $path = drupal_get_path('module', 'locationmap');
 
-  drupal_add_js('//maps.google.com/maps/api/js?v=3',  array('type' => 'external', 'weight' => 5));
+  $locationmap_api_key = variable_get('locationmap_api_key', '');
+
+  if($locationmap_api_key) {
+    drupal_add_js('//maps.google.com/maps/api/js?v=3&key=' . $locationmap_api_key,  array('type' => 'external', 'weight' => 5));
+  }
+  else {
+    drupal_add_js('//maps.google.com/maps/api/js?v=3',  array('type' => 'external', 'weight' => 5));
+    if(user_access('administer locationmap')) {
+      $msg = l(t('Location Map Configuration Page'), 'admin/config/content/locationmap');
+      $google_map_key_msg = l(t('Google API Console'), 'https://developers.google.com/maps/documentation/javascript/get-api-key');
+      drupal_set_message(t("Please set Google Map API Key at $msg. You can get Google Map API key from $google_map_key_msg"), 'warning', FALSE);
+    }
+  }
   drupal_add_js($path . '/locationmap.js', array('type' => 'file', 'weight' => 6, 'scope' => 'footer'));
 
   $locationmap_css_options = array(
@@ -479,6 +500,9 @@ function locationmap_block_view($delta) {
  */
 function locationmap_static_image_url($w = 200, $h = 150) {
   $image_url = '//maps.googleapis.com/maps/api/staticmap?zoom=' . variable_get('locationmap_zoom', 5);
+  if ($locationmap_api_key = variable_get('locationmap_api_key', '')) {
+    $image_url .= '&amp;key=' . $locationmap_api_key;
+  }
   $image_url .= "&amp;size={$w}x{$h}";
   $image_url .= '&amp;markers=' . variable_get('locationmap_lat', 0) . ',' . variable_get('locationmap_lng', 0);
   return($image_url);
@@ -502,7 +526,19 @@ function locationmap_block_image() {
       'format' => filter_default_format(),
     );
     $locationmap_info = variable_get('locationmap_info', $locationmap_info_defaults);
-    drupal_add_js('//maps.google.com/maps/api/js?v=3',  array('type' => 'external', 'weight' => 5));
+    $locationmap_api_key = variable_get('locationmap_api_key', '');
+
+    if($locationmap_api_key) {
+      drupal_add_js('//maps.google.com/maps/api/js?v=3&key=' . $locationmap_api_key,  array('type' => 'external', 'weight' => 5));
+    }
+   else {
+      drupal_add_js('//maps.google.com/maps/api/js?v=3',  array('type' => 'external', 'weight' => 5));
+      if(user_access('administer locationmap')) {
+        $msg = l(t('Location Map Configuration Page'), 'admin/config/content/locationmap');
+        $google_map_key_msg = l(t('Google API Console'), 'https://developers.google.com/maps/documentation/javascript/get-api-key');
+        drupal_set_message(t("Please set Google Map API Key at $msg. You can get Google Map API key from $google_map_key_msg"), 'warning', FALSE);
+      }
+    }
     drupal_add_js($path . '/locationmap.js', array('type' => 'file', 'weight' => 6, 'scope' => 'footer'));
     drupal_add_css($path . '/locationmap.css');
     $locationmap_settings = array(
-- 
2.9.0

